File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int advancedBinarySearch(vector<int>& nums, int target)
    {
        int l=0,h=nums.size();
        while(l<h)
        {
            int mid=l+(h-l)/2;
            if(nums[mid]<target)
            l=mid+1;
            else
            h=mid;
        }
        return l;
    }
    vector<int> searchRange(vector<int>& nums, int target) 
    {
        int l=advancedBinarySearch(nums,target);
        
        if(l==nums.size() || nums[l]!=target)// && h!=nums.size()-1)
        return {-1,-1};
        
        int h=advancedBinarySearch(nums,target+1);
        return {l,h-1};        
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int advBinSearch(int nums[],int target)
    {
        int l=0,h=nums.length;

        while(l<h)
        {
            int mid=l+(h-l)/2;

            if(nums[mid]<target)
            l=mid+1;
            else
            h=mid;
        }
        return l;
    }
    public int[] searchRange(int[] nums, int target) {
        
        int l=advBinSearch(nums,target);
        System.out.println(l);
        if(l==nums.length || nums[l]!=target)
        return new int[]{-1,-1};

        int h=advBinSearch(nums,target+1);
        System.out.println(h);
        // if(h==nums.length || nums[h-1]!=target)
        // return new int[]{-1,-1};
        
        // else
        return new int[]{l,h-1};
    }
}